utils/stats.ts

```diff
@@ -203,31 +225,67 @@ export const getBattleStats = (...battles: BattleBrief[]): BattleStats => {
     all = battle.otherTeams
       .flat()
       .reduce((prev, current) => addBattlePlayerStats(prev, current), all);
+
     if (!stageMap.has(battle.stage)) {
       stageMap.set(battle.stage, new Map());
     }
     if (!stageMap.get(battle.stage)!.has(battle.rule)) {
-      stageMap.get(battle.stage)!.set(battle.rule, { count: 0, win: 0 });
+      stageMap.get(battle.stage)!.set(battle.rule, { count: 0, win: 0, weaponMap: new Map() });
+    }
+    if (!stageMap.get(battle.stage)!.get(battle.rule)!.weaponMap.has(selfPlayerBrief.weapon)) {
+      stageMap
+        .get(battle.stage)!
+        .get(battle.rule)!
+        .weaponMap.set(selfPlayerBrief.weapon, { count: 0, win: 0 });
     }
     const stage = stageMap.get(battle.stage)!.get(battle.rule)!;
+    const stageWeapon = stage.weaponMap.get(selfPlayerBrief.weapon)!;
     stage.count += 1;
-    stage.win += battle.result === Judgement.WIN ? 1 : 0;
+    stageWeapon.count += 1;
+    if (battle.result === Judgement.WIN) {
+      stage.win += 1;
+      stageWeapon.win += 1;
+    }
+
     if (!weaponMap.has(selfPlayerBrief.weapon)) {
       weaponMap.set(selfPlayerBrief.weapon, new Map());
     }
     if (!weaponMap.get(selfPlayerBrief.weapon)!.has(battle.rule)) {
-      weaponMap.get(selfPlayerBrief.weapon)!.set(battle.rule, { count: 0, win: 0 });
+      weaponMap
+        .get(selfPlayerBrief.weapon)!
+        .set(battle.rule, { count: 0, win: 0, stageMap: new Map() });
+    }
+    if (!weaponMap.get(selfPlayerBrief.weapon)!.get(battle.rule)!.stageMap.has(battle.stage)) {
+      weaponMap
+        .get(selfPlayerBrief.weapon)!
+        .get(battle.rule)!
+        .stageMap.set(battle.stage, { count: 0, win: 0 });
     }
     const weapon = weaponMap.get(selfPlayerBrief.weapon)!.get(battle.rule)!;
+    const weaponStage = weapon.stageMap.get(battle.stage)!;
     weapon.count += 1;
-    weapon.win += battle.result === Judgement.WIN ? 1 : 0;
+    weaponStage.count += 1;
+    if (battle.result === Judgement.WIN) {
+      weapon.win += 1;
+      weaponStage.win += 1;
+    }
   }
   const stages = Array.from(stageMap, (stage) => {
-    const rules = Array.from(stage[1], (rule) => ({
-      id: rule[0],
-      count: rule[1].count,
-      win: rule[1].win,
-    }));
+    const rules = Array.from(stage[1], (rule) => {
+      const weapons = Array.from(rule[1].weaponMap, (weapon) => ({
+        id: weapon[0],
+        count: weapon[1].count,
+        win: weapon[1].win,
+      }));
+      // sort weapon by win rate (`count` won't be 0)
+      weapons.sort((a, b) => b.win / b.count - a.win / a.count);
```

Comment:
sorted by win rate, I find it useful for analyzing the weapon performance, I can change it to sort by name if you prefer that.

Reply:
Oh sorry if I didn't make it clear. The sorting is only applied to the new level of lists introduced in this PR. The order of the existing lists remains unchanged.

Just to confirm, if you prefer to keep the order consistent with the existing lists, I can make the change.

Reply:
I prefer to start with ID sorting only. And then we can add a segment control letting users decide how they want to sort the list (by ID, by usage or by win rate).